File: 2023/advent_of_code/test/q10/q10_test.exs
defmodule AdventOfCode.Q10Test do
  use ExUnit.Case
  doctest AdventOfCode.Q10

  test "Q10 Part 1 Example works" do
    file_stream = File.stream!("./test/q10/q10_example.txt")
    result = AdventOfCode.Q10.part1(file_stream)
    assert result == 4
  end

  test "Q10 Part 1 Example 2 works" do
    file_stream = File.stream!("./test/q10/q10_example2.txt")
    result = AdventOfCode.Q10.part1(file_stream)
    assert result == 4
  end

  test "Q10 Part 1 Example 3 works" do
    file_stream = File.stream!("./test/q10/q10_example3.txt")
    result = AdventOfCode.Q10.part1(file_stream)
    assert result == 8
  end

  test "Q10 Part 1 works" do
    file_stream = File.stream!("./test/q10/q10.txt")
    result = AdventOfCode.Q10.part1(file_stream)
    assert result == 6820
  end

  test "Q10 Part 2 Example works" do
    file_stream = File.stream!("./test/q10/q10_example4.txt")
    result = AdventOfCode.Q10.part2(file_stream)
    assert result == 4
  end

  test "Q10 Part 2 Example 2 works" do
    file_stream = File.stream!("./test/q10/q10_example5.txt")
    result = AdventOfCode.Q10.part2(file_stream)
    assert result == 4
  end

  test "Q10 Part 2 Example 3 works" do
    file_stream = File.stream!("./test/q10/q10_example6.txt")
    result = AdventOfCode.Q10.part2(file_stream)
    assert result == 8
  end

  test "Q10 Part 2 Example 4 works" do
    file_stream = File.stream!("./test/q10/q10_example7.txt")
    result = AdventOfCode.Q10.part2(file_stream)
    assert result == 10
  end

  test "Q10 Part 2" do
    file_stream = File.stream!("./test/q10/q10.txt")
    result = AdventOfCode.Q10.part2(file_stream)
    assert result == 337
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q10/Q10Test.kt
package q10

import java.nio.file.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class Q10Test {
    @Test
    fun `part 1 works`() {
        val filePath = javaClass.getResource("/q10.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(14340, part1(filePath!!))
    }

    @Test
    fun `part 2 works`() {
        val filePath = javaClass.getResource("/q10.txt")?.toURI()?.let { Paths.get(it) }
        // PAJCBHP
        val expected = """
            ###...##..###....##..##..###..#..#.###..
            #..#.#..#.#..#....#.#..#.#..#.#..#.#..#.
            #..#.#..#.#..#....#.#....###..####.#..#.
            ###..####.###.....#.#....#..#.#..#.###..
            #....#..#.#....#..#.#..#.#..#.#..#.#....
            #....#..#.#.....##...##..###..#..#.#....
        """.trimIndent()
        assertEquals(expected, part2(filePath!!))
    }
}